x-pack/plugins/infra/public/pages/metrics/inventory_view/hooks/use_waffle_options.ts

```diff
@@ -121,6 +122,13 @@ export const useWaffleOptions = () => {
     [setState]
   );
 
+  const { inventoryPrefill } = useAlertPrefillContext();
+  useEffect(() => {
+    const { setNodeType, setMetric } = inventoryPrefill;
+    setNodeType(state.nodeType);
+    setMetric(state.metric);
+  }, [state, inventoryPrefill]);
+
   return {
```

Comment:
We should add a test for this, unfortunately we don't have one setup yet but there's no time like the present :D. I would look at `x-pack/plugins/infra/public/pages/metrics/metrics_explorer/hooks/use_metrics_explorer_options.test.tsx` for inspiration.